### PR TITLE
Update Documentation & Clean-Up Code for Redesign

### LINKED ISSUE #337: Update Documentation Prior to Redesign Merge
Before merging the redesign into main, documentation should be appropriatly updated to account for the redesign.

### PR DESCRIPTION
**Pull request by philipc2**
Closes #337, #247

[Documentation ](https://uxarray--336.org.readthedocs.build/en/336/)

## Overview

* Major refactor to documentation in preparation for merging the Redesign
   - Getting Started Guide (Overview, Quick Install, & FAQ)
   - Overhauled User & Internal API Reference to better reflect our core data structures (Grid, UxDataset, UxDataArray)
   - Updated USage Example Notebooks 
   - Updated Docstrings for all functions 
* Miscellaneous 
   - Fixed `__repr__` not properly printing dimensions 

## PR Checklist

**General**
- [x] An issue is linked created and linked
- [x] Add appropriate labels
- [x] Filled out Overview and Expected Usage (if applicable) sections

**Testing**
- [x] Adequate tests are created if there is new functionality
- [x] Tests cover all possible logical paths in your function
- [x] Tests are not too basic (such as simply calling a function and nothing else)

**Documentation**
- [x] Docstrings have been added to all new functions
- [x] Docstrings have updated with any function changes
- [x] Internal functions have a preceding underscore (`_`) and have been added to `docs/internal_api/index.rst`
- [x] User functions have been added to `docs/user_api/index.rst`

**Examples**
- [x] Any new notebook examples added to `docs/examples/` folder
- [x] Clear the output of all cells before committing
- [x] New notebook files added to `docs/examples.rst` toctree
- [x] New notebook files added to new entry in `docs/gallery.yml` with appropriate thumbnail photo in `docs/_static/thumbnails/`

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -5,6 +5,8 @@
 API Reference
 =============
 
+
+
 Below user API and internal API pages show already-implemented Uxarray functionality.
 You can also check the draft `Uxarray API
 <https://github.com/UXARRAY/uxarray/blob/main/docs/user_api/uxarray_api.md>`_
```
Comment (aaronzedwick) [inline docs/api.rst:12]:
```suggestion
Below user API and internal API pages show already-implemented UXarray functionality.
You can also check the draft `UXarray API
<https://github.com/UXARRAY/uxarray/blob/main/docs/user_api/uxarray_api.md>`_
```

File: docs/index.rst [hunk 2]
```diff
@@ -24,8 +24,12 @@
 Uxarray Documentation
```
Comment (aaronzedwick) [inline docs/index.rst:24]:
```suggestion
UXarray Documentation
```

File: docs/internal_api/index.rst [hunk 3]
```diff
@@ -1,36 +1,99 @@
 .. currentmodule:: uxarray
 
+########
 Internal API
-============
+########
 
 This page shows already-implemented Uxarray internal API functions. You can also
 check the draft `Uxarray API
```
Comment (aaronzedwick) [inline docs/internal_api/index.rst:8]:
```suggestion
check the draft `UXarray API
```

File: docs/installation.rst [hunk 4]
```diff
@@ -59,7 +59,7 @@ how to setup a conda environment with them.
 Creating a Conda environment
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-The Uxarray source code includes a conda environment definition file
+The UXarray source code includes a conda environment definition file
 (:code:`environment.yml`) in the :code:`/ci` folder under the root
 directory that can be used to create a development environment
 containing all of the packages required to build Uxarray. The
```
Comment (aaronzedwick) [inline docs/installation.rst:65]:
```suggestion
containing all of the packages required to build UXarray. The
```

**Review by aaronzedwick — COMMENTED** [review]
From what I can see looks good, just some places you forgot to replace Uxarray with UXarray.

**Comment by aaronzedwick** [comment]
Oh also, I didn't see any changes made to the "How to Cite Uxarray" page, which also has some Uxarray usage.

**Review by erogluorhan — APPROVED** [review]

**Pull request merged**